AsyncDisplayKit/ASPagerNode.m

```diff
@@ -0,0 +1,74 @@
+//
+//  ASPagerNode.m
+//  AsyncDisplayKit
+//
+//  Created by Levi McCallum on 12/7/15.
+//  Copyright © 2015 Facebook. All rights reserved.
+//
+
+#import "ASPagerNode.h"
+
+#import <AsyncDisplayKit/AsyncDisplayKit.h>
+
+@interface ASPagerNode () <ASCollectionViewDataSource, ASCollectionViewDelegateFlowLayout> {
+  UICollectionViewFlowLayout *_flowLayout;
+}
+
+@end
+
+@implementation ASPagerNode
+
+- (instancetype)init
+{
+  _flowLayout = [[UICollectionViewFlowLayout alloc] init];
+  _flowLayout.scrollDirection = UICollectionViewScrollDirectionHorizontal;
+  _flowLayout.minimumInteritemSpacing = 0;
+  _flowLayout.minimumLineSpacing = 0;
+  
+  self = [super initWithCollectionViewLayout:_flowLayout];
+  if (self != nil) {
+    self.view.asyncDataSource = self;
```

Comment:
It is essential that we set these in -didLoad, not in -init.  Some big perf problems in Pinterest were addressed by ensuring we didn't poke the creation of ASCollectionViews too early.